de/nmichael/efa/gui/AdminDialog.java

```diff
@@ -243,6 +245,13 @@ public void keyAction(ActionEvent evt) {
         _keyAction(evt);
     }
 
+    public void closeButton_actionPerformed(ActionEvent e) {
+        if ((Daten.project.getProjectStorageType() == IDataAccess.TYPE_EFA_CLOUD)
+                && (TxRequestQueue.getInstance() != null))
+            TxRequestQueue.getInstance().clearAdminCredentials();
+        cancel();
```

Comment:
@tfyh Thanks for making this change.

The proper way of doing it in OO design is to override it (as you've now done) and (unless you want to completely replace the original implementation) then invoke the inherited method. So, instead of calling "cancel()" here, please replace this with super.closeButton_actionPerformed(e). This will ultimately do the same thing (it will call BaseDialog.closeButton_actionPerformed(e), which will call cancel(), but if the implementation in BaseDialog ever does something more, AdminDialog would automatically inhertit it.

A good convention is also to annotate overridden methods with
@Overrive
above line 248 above. It's something I also unfortunately didn't always do, but it's recommended to make clear that this is actually overriding an inherited method, and forces the compiler to check that you actually do override it. For example, if the method signature in BaseDialog.java would ever change, then without the @Override the compiler wouldn't detect that you no longer override this method, and the consequence would be that your closeButton_actionPerformed in AdminDialog would no longer be called by anyone. The @Override would result in a compiler error in this case and thus surface this immediately.

So the method should ultimately be:
```
    @Override
    public void closeButton_actionPerformed(ActionEvent e) {
        if ((Daten.project.getProjectStorageType() == IDataAccess.TYPE_EFA_CLOUD)
                && (TxRequestQueue.getInstance() != null))
            TxRequestQueue.getInstance().clearAdminCredentials();
        super.closeButton_actionPerformed(e);
    }
```